src/scripts/buildMetadata.ts

```diff
@@ -63,26 +63,28 @@ export function buildMetadata(
             adaptersController.fetchChainProtocolAdapters(chainId, protocolId)
 
           for (const [_, adapter] of chainProtocolAdapters) {
-            if (isIMetadataBuilder(adapter)) {
-              const { metadata, fileDetails } = (await adapter.buildMetadata(
-                true,
-              )) as {
-                metadata: Json
-                fileDetails: {
-                  protocolId: Protocol
-                  productId: string
-                  chainId: Chain
-                  fileKey: string
-                }
+            if (!isIMetadataBuilder(adapter)) {
+              continue
+            }
+
+            const { metadata, fileDetails } = (await adapter.buildMetadata(
+              true,
+            )) as {
+              metadata: Json
+              fileDetails: {
+                protocolId: Protocol
+                productId: string
+                chainId: Chain
+                fileKey: string
               }
+            }
 
-              await writeMetadataToFile({
-                ...fileDetails,
-                metadata,
-              })
+            await writeMetadataToFile({
+              ...fileDetails,
+              metadata,
+            })
 
-              addStaticImport(fileDetails)
-            }
+            await addStaticImport(fileDetails)
```

Comment:
This was the first issue that was causing `AdaptersMetadata.ts` file to become corrupt, since we were not awaiting, sometimes two process would write at the same time.